Develop

SergeiTrofimov opened the pull request:
sprint_4.
финальное задание 
два браузера

Commit:
1.Добавлены тесты для списка вопросов
2.Тесты для заказов Chrome и Mozilla

Commit:
1. Добавлена параметризация
2. Добавлены комментарии
3. Убран мусор

Commit:
1. Добавлена параметризация
2. Добавлены комментарии
3. Убран мусор

Commit:
1. Добавлена параметризация
2. Добавлены комментарии
3. Убран мусор

nikolay-977 commented on src/test/java/ru/Constants.java:
```diff
@@ -0,0 +1,10 @@
+
+public class Constants { // тут хранятся константы для тестов
+    public static final String hostname = "https://qa-scooter.praktikum-services.ru/";
+    public static final String accordionHeaderNamePrefix = "accordion__heading-";
```
Локаторы следует описывать в классах page objects

nikolay-977 commented on src/test/java/ru/praktikum/qa/pageobjects/OrderPage.java:
```diff
@@ -0,0 +1,116 @@
+public class OrderPage extends MainAbstratPage {
+
+    // Локаторы первой страницы
+    private By orderNextRegButtonPage1 = By.xpath("//*[@id=\"root\"]/div/div[2]/div[3]/button"); // Локатор кнопки "Дальше"
```
Локаторы не следует описывать через цепочку div элементов с индексами. Такие локаторы сложно поддерживать. Локатор должен быть легко читаем. Это сократит время локализации дефекта.

nikolay-977 commented on src/test/java/ru/praktikum/qa/OrderScooterTestChrome.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    // параметры теста
+ @Parameterized.Parameters
```
В параметризованных тестах для аннотации @Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки.

nikolay-977 commented on src/test/java/ru/praktikum/qa/OrderScooterTestChrome.java:
```diff
@@ -0,0 +1,70 @@
+    { "Лев", "Зайцев", "Пушкин","79112489001","13.03.2023","Оставить у дверей",1,1},
+     { "Иван", "Болван", "Горький","+79112489002","24.05.2022","Не звонить в дверь",3,1},
+     };
+  }
```
Используй сочетание клавиш Ctrl+Alt+L (Option+Command+L для Mac) для расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea.

nikolay-977 commented on src/test/java/ru/praktikum/qa/OrderScooterTestFirefox.java:
```diff
@@ -0,0 +1,71 @@
+        this.metroStationIndex = metroStationIndex;
+        this.rentDurationIndex = rentDurationIndex;
+    }
+
```
Code Style. Убери лишние пропуски строк. Для зрительного разделения кода достаточно использовать один пропуск строки. Между фигурными скобками пропуски не нужны. Пропуски строк должны быть однообразными.

nikolay-977 commented on src/test/java/ru/praktikum/qa/QuestionsListTestChrome.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    @Test // перебор вопросов (открываем все вопросы, сравниваем описания)
+    public void accordionElementShouldHaveDescriptionAll() {
```
Лучше сделать через параметризацию

Commit:
1. Добавлена параметризация в тест вопросов для Chrome
2. Добавлена аннотация name в параметризацию
3. Перенесены локаторы из констант в pageobject
4. Исправлены длинные локаторы
5. Исправлено форматирование кода
6. Убраны лишние пустые строки

Commit:
1. Убран еще одна провисающая строка
